executor/explain_test.go

```diff
@@ -326,3 +327,20 @@ func (s *testSuite1) TestCheckActRowsWithUnistore(c *C) {
 		checkActRows(c, tk, test.sql, test.expected)
 	}
 }
+
+func (s *testSuite2) TestExplainAnalyzeCTEMemoryAndDiskInfo(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t (a int)")
+	tk.MustExec("insert into t with recursive cte(a) as (select 1 union select a + 1 from cte where a < 1000) select * from cte;")
+
+	tk.MustExec("set @@tidb_mem_quota_query=10240;")
+	rows := tk.MustQuery("explain analyze with recursive cte(a) as (select 1 union select a + 1 from cte where a < 1000)" +
+		" select * from cte, t;").Rows()
+
```

Comment:
```suggestion
checkPlan := func(rows [][]interface{}, expectPlan string) {
		buf := bytes.NewBufferString("")
		for _, row := range rows {
			for i := range row {
				fmt.Fprintf(buf, "%v ", row[i])
			}
		}
		c.Assert(buf.String(), Matches, expectPlan, Commentf("%v not match %v", buf.String(), expectPlan))
	}
	expectedPlan := ".*HashJoin.*CTEFullScan.*TableReader.*TableFullScan.*CTE.*Projection.*Concurrency.*TableDual.*Projection.*Concurrency.*Selection.*CTETable.*"
	checkPlan(rows, expectedPlan)
	
```

Reply:
Add test for the plan tree.

Reply:
plan tree has been test in pr https://github.com/pingcap/tidb/pull/24986